lib/vfscore: Fix build warning for argument declaration

calinstefan025 opened the pull request:
Resolve a build warning in the `vfscore` library caused by the declaration of argument 2 as a `const struct timeval *` pointer. This change ensures compatibility with the expected parameter type.

<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [ ] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.


### Base target

 - Architecture(s): [e.g. `x86_64` or N/A]
 - Platform(s): [e.g. `kvm`, `xen` or N/A]
 - Application(s): [e.g. `app-python3` or N/A]


### Additional configuration

<!--
Please specify any additional configuration which is needed for this feature to
work or any new configuration parameters which are introduced by this PR.  This
will help during the review process.  For example:

 - `CONFIG_LIBUKDEBUG=y`

-->

### Description of changes
Changed `const struct timeval *times` parameter to `const struct timeval times[2]` in the `futimes` and `lutimes` functions to resolve the argument declaration warning.

<!--
Please provide a detailed description of the changes made in this new PR.
-->

calinstefan025 committed:
lib/vfscore: Fix build warning for argument declaration

Resolve a build warning in the `vfscore` library caused by the
declaration of argument 2 as a `const struct timeval *` pointer.
This change ensures compatibility with the expected parameter type.

Signed-off-by: Calin Avramoniu <clavramoniu@gmail.com>

mariapana reviewed (approved):
LGTM, thanks!

Reviewed-by: Maria Pana <maria.pana4@gmail.com>

razvand reviewed (changes requested):
Other function names should be updated for consistency:

- [`sys_utimes` declaration `lib/vfscore/vfs.h`](https://github.com/unikraft/unikraft/blob/staging/lib/vfscore/vfs.h#L464)
- [`utimes`, `futimes`, `futimesat` and `lutimes` declarations in `lib/nolibc/musl-imported/include/sys/time.h`](https://github.com/unikraft/unikraft/blob/staging/lib/nolibc/musl-imported/include/sys/time.h)

calinstefan025 commented:
Renaming the functions create other warnings like this:
```console
/home/calin/catalog-core/repos/unikraft/lib/vfscore/syscalls.c:1329:46: warning: argument 2 of type ‘const struct timeval *’ declared as a pointer [-Warray-parameter=]
 1329 | sys_utimes(char *path, const struct timeval *times, int flags)
      |                        ~~~~~~~~~~~~~~~~~~~~~~^~~~~
In file included from /home/calin/catalog-core/repos/unikraft/lib/vfscore/syscalls.c:58:
/home/calin/catalog-core/repos/unikraft/lib/vfscore/vfs.h:480:49: note: previously declared as an array ‘const struct timeval[2]’
  480 | int sys_utimes(char *path, const struct timeval times[2], int flags);
```

Should I fix those too?

razvand commented:
> Renaming the functions create other warnings like this:
> 
> ```
> /home/calin/catalog-core/repos/unikraft/lib/vfscore/syscalls.c:1329:46: warning: argument 2 of type ‘const struct timeval *’ declared as a pointer [-Warray-parameter=]
>  1329 | sys_utimes(char *path, const struct timeval *times, int flags)
>       |                        ~~~~~~~~~~~~~~~~~~~~~~^~~~~
> In file included from /home/calin/catalog-core/repos/unikraft/lib/vfscore/syscalls.c:58:
> /home/calin/catalog-core/repos/unikraft/lib/vfscore/vfs.h:480:49: note: previously declared as an array ‘const struct timeval[2]’
>   480 | int sys_utimes(char *path, const struct timeval times[2], int flags);
> ```
> 
> Should I fix those too?

Yes. They should also be renamed, for consistency throughout the code and with Musl.

calinstefan025 committed:
lib/{nolibc,vfscore}: Update function declarations

Updated the requested function declarations to maintain
consistency throughout the code.

Signed-off-by: Calin Avramoniu <clavramoniu@gmail.com>

calinstefan025 commented:
I changed the declarations where it was suggested and fixed the build warnings.

razvand reviewed (changes requested):
Use `lib/{nolibc,vfscore}:` as the prefix for the second commit. Add a commit description to detail the rationale behind the change.

Use the format `Firstname Lastname <email>` for both the commit author and for the `Signed-off-by:` part.

calinstefan025 commented:
Hey @razvand, I changed the commits like you requested.
